fluence-js/3_in_depth.md

```diff
@@ -350,3 +350,177 @@ Tetraplets have the form of:
 To learn more about tetraplets and application security see [Security](https://github.com/fluencelabs/gitbook-docs/tree/77344eb147c2ce17fe1c0f37013082fc85c1ffa3/js-sdk/knowledge_security.md)
 
 To see full specification of `CallParams` type see [API reference](https://github.com/fluencelabs/gitbook-docs/tree/77344eb147c2ce17fe1c0f37013082fc85c1ffa3/js-sdk/js-sdk/6_reference/modules.md)
+
+### Signing service
+
+Signing service is useful when you want to sign arbitrary data and pass it further inside a singe aqua script. Signing service allows to restrict it's usage for security reasons: e.g you don't want to sign anything except it comes from a trusted source. The aqua side API is the following:
```

Comment:
singe -> single
What's the significance of 'single' aqua script?
restrict it's usage  -> its; it's as in the arbitrary data ?
e.g you don't want to sign anything except <if/when > it comes from a trusted source. -> don't quite follow: data comes from a trusted source, then i sign it, then it's used in an aqua script ... ??